cogs/timer.py

```diff
@@ -2174,6 +2174,168 @@ async def permissionCheck(self, msg, author):
+                return start,dmChar #cause error
+            elif rewardUser == dmChar[0]: 
+                userFound = True
+                # the player entry of the player getting the item
+                currentItem = dmChar
+                # list of current consumables on the character
+                # [1] in a player entry is the DB entry of the character
+                charConsumableList = currentItem[1]['Consumables'].split(', ')
+                # list of current magical items
+                charMagicList = currentItem[1]['Magic Items'].split(', ')
+                # character level
+                charLevel = int(currentItem[1]['Level'])
+            # since this checks for multiple things, this cannot be avoided
+            for u, v in startcopy.items():
+
+                for item in v:
+                    if dmChar[0] == rewardUser:
+                        break
+                    if item[0] == rewardUser:
+                        userFound = True
+                        # the player entry of the player getting the item
+                        currentItem = item
+                        # list of current consumables on the character
+                        # [1] in a player entry is the DB entry of the character
+                        charConsumableList = currentItem[1]['Consumables'].split(', ')
+                        # list of current magical items
+                        charMagicList = currentItem[1]['Magic Items'].split(', ')
+                        # character level
+                        charLevel = int(currentItem[1]['Level'])
+        
+        
+        # calculate the tier of the rewards
+        tierNum = 5
+        if charLevel < 5:
+            tierNum = 1
+        elif charLevel < 11:
+            tierNum = 2
+        elif charLevel < 17:
+            tierNum = 3
+        elif charLevel < 20:
+            tierNum = 4
+        
+        rewardCollection = db.rit
+        rewardTable = list(rewardCollection.find({"Tier": tierNum, "Minor/Major": rewardType})) #gets the full list of eligible rewards based on input
+        
+        randomItem = random.choice(rewardTable) #picks a random reward
+        rewardString = []
+        
+        rewardConsumable_group_type = "Name"
+        if "Grouped" in randomItem: # used to check for duplicates if the random reward has subtypes, like ammunition
+           rewardConsumable_group_type = "Grouped"
+        
+        player_type = "Players"
+        if rewardUser == dmChar[0]:
+            player_type = "DM"
+        
+        rewardCheck = randomItem
+        
+        # check if the item has already been rewarded to the players
+        if (rewardCheck[rewardConsumable_group_type] in dmChar[5][1][player_type]["Major"] or
+            rewardCheck[rewardConsumable_group_type] in dmChar[5][1][player_type]["Minor"]):
+            return None # returns none, which will cause the command to be repeated to try to get a non-duplicated reward item
+        
```

Comment:
There are 11 possible T5 minors, Eternal Noodle can hand out 8 of them, by running a 15h game one could hand out all possible T5 minors and then use this command and deadlock because all options have been rewarded. In any case it basically turns into a gamble on how long this command runs and that is unacceptable. There are ways to avoid the issue, though.